### ISSUE TITLE
Optionally ignore logical-or in complexity rule

### ISSUE THREAD

**Issue by Stuk** [eslint/eslint#3473]
**Problem**

We have some code that has a large number of fallbacks in the form:

```js
options = options || {};
thing = options.thing || "default";
```

This increases the complexity in the [`complexity` rule](https://github.com/eslint/eslint/blob/5c6d6de75359d887196bcf7636baa47909025c52/lib/rules/complexity.js) to the point where we get a lot of extraneous warnings, however it seems to me that this doesn't make the code much harder to reason about, as the branches are very short lived.

**Solution**

Is there any interest in me implementing a new option for the complexity rule that causes `||` used in this way to be ignored?

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by nzakas** [comment]
@Stuk I'm not sure I understand the rationale here - logical OR is a factor of complexity, so ignoring them will give you a bad overview of how complex your code really is. 

Cyclomatic complexity isn't about how hard it is to reason about. From [Wikipedia](https://en.wikipedia.org/wiki/Cyclomatic_complexity):

>  It is a quantitative measure of the number of linearly independent paths through a program's source code.

If you really want this, I'd recommend creating a custom rule by copying the existing one and modifying. I don't think altering the calculation of cyclomatic complexity would appeal to a large audience.

**Comment by Stuk** [comment]
The rationale is that frequently `||` is used to provide default values for function arguments. I assume the new ES default function arguments won't count against cyclomatic complexity, and so neither should `||`.

However I understand that "cyclomatic complexity" has a defined meaning, and so changing it on a whim doesn't make sense.

Thank you for your reply!

**Comment by keithkml** [comment]
This comes up a lot in our codebase. I imagine most developers find the current eslint behavior unexpected and undesirable.
